wp-media/wp-rocket
# Improve support for Web Stories plugin

Improves support for Web Stories for WordPress by checking for `is_amp_endpoint` at a later stage.

This way, WP Rocket will support AMP documents as output by the official [AMP plugin for WordPress] as well as [Web Stories for WordPress](https://wp.stories.google/), which both support the `is_amp_endpoint` function checks.

However, in the case of Web Stories, `is_amp_endpoint` is only defined on the `wp` action as a polyfill for cases like this, and is not defined earlier. Hence doing the `function_exists` check at this stage instead of in the `get_subscribed_events()` method.

To test:

1. Install Web Stories for WordPress
1. Create a web story
1. Verify that `Minification, DNS Prefetch, LazyLoad, Defer JS` are not active when viewing the web story.

Fixes #3102
